packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -130,6 +131,16 @@ function createComponentDef(Ctor: LightningElementConstructor): ComponentDef {
             );
         }
 
+        // TODO [#3971]: Completely remove shadowSupportMode "any"
+        if (
+            !isUndefined(ctorShadowSupportMode) &&
+            ctorShadowSupportMode === ShadowSupportMode.Any
```

Comment:
```suggestion
            ctorShadowSupportMode === ShadowSupportMode.Any
```

It's not necessary to test for undefined-ness here.